channels/ON-CHAIN.md

```diff
@@ -0,0 +1,372 @@
+for. It is advised that peers choose an even amount, otherwise one peer will end
+up paying more, although this should not be an issue in practice.
+
+```
+if initiator_amount + responder_amount < fee
+  return error
+else if initiator_amount >= ceil(fee/2) && responder_amount >= floor(fee/2)
+  initiator_final := initiator_amount - ceil(fee/2)
+  responder_final := responder_amount - floor(fee/2)
+else if responder_amount >= ceil(fee/2) && initiator_amount >= floor(fee/2)
+  responder_final := responder_amount - ceil(fee/2)
+  initiator_final := initiator_amount - floor(fee/2)
+else if initiator_amount > responder_amount
+  initiator_final := initiator_amount - fee + responder_amount
+  responder_final := 0.0
+else
+  responder_final := responder_amount - fee + initiator_amount
+  initiator_final := 0.0
+```
+
+
+
+#### Requirements
+
+This transaction MUST have valid signatures of all involved parties.
+
+After this transaction has been included in a block, the channel MUST be
+considered closed and allow no further modifications.
+
+
+### `channel_close_solo`
+
+In order to close a channel unilaterally, a user has to send a
+`channel_close_solo` transaction. This is only necessary if one peer stops
+responding but can also be used by an malicious peer trying to close a channel
+with a state that hasn't been agreed on by all participants.
+
+With the inclusion of this transaction on chain, the timer, during which
+disputes in the form of `channel_slash` will be considered, is started.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| amount               | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`:
+- `from`
+- `ttl`
+- `nonce`
+- `fee`
+
+
+### `channel_slash`
+
+If a malicious party sent a `channel_close_solo` with an outdated state, the
+honest party has the opportunity to issue a `channel_slash` transaction. This
+transaction needs to include a state signed by all peers with a higher sequence
+number.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| amount               | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`:
+- `ttl`
+- `nonce`
+- `fee`
+
+
+#### Requirements
+
+MUST be signed using private key corresponding to the public key `from`.
+
+### `channel_settle`
+
+The settlement transaction is the last one in the lifecycle of a channel, but
```

Comment:
Do I understand correctly that if no channel_settle is sent then channel stays in state tree forever in state closing?

Reply:
Yes, but parties have an incentive of closing it as tokens are otherwise locked in a dysfunctional channel. Closing it is the very idea of posting a `solo_close` transaction in the first place.